File: myext.py
import numpy as np
import myext
import numpy as np
import inspect

def make_vec():
    n = 4
    m = 3
    a1 = np.array(n)
    a1 = []
    for i in range(n):
       a1.append(i+1)
    a1 = np.array(a1)
    b1 = []
    for j in range(m):
       b1.append(j+1)
    b1 = np.array(b1)
    return a1,b1


def test1():
    print 
    print "subroutine",inspect.currentframe().f_code.co_name
    a1,b1 = make_vec()
    print "a=",a1
    print "b=",b1
    print "apply a*b"
    c1 = myext.m_mylib.vavbmc(a1,b1)
    print "c=",c1
    print 


def make_matrix():
    n = 4
    a1 = myext.m_mylib.init("+",n,0.1)
    b1 = myext.m_mylib.init("-",n,0.1)
    return a1,b1

def test2():
    print 
    print "subroutine",inspect.currentframe().f_code.co_name
    a1,b1 = make_matrix()
    print "a=",a1
    print "b=",b1
    print "apply a+b"
    c1 = myext.m_mylib.apb(a1,b1)
    print "c1=",c1

def test3():
    print
    print "subroutine",inspect.currentframe().f_code.co_name
    a1,b1 = make_matrix()
    print "a=",a1
    print "b=",b1
    p = 10.0 
    print "apply (a+b)*p, p=",p
    c1 = myext.m_mylib.apb_p(p, a1,b1)
    print "c1=",c1


if __name__ == "__main__":
    test1()
    test2()
    test3()

File: myext.f90


module m_mylib 
contains

subroutine init(mode,n,fac,a)
use m_ext
implicit none
character,intent(in)::mode
integer,intent(in)::n
real(8),intent(in):: fac
real(8),intent(out):: a(n,n)

call m_ext_allocate(n)
call m_ext_make_g_v(mode)

a = g_v * fac 
end subroutine init


subroutine vavbmc(n,m,a,b,c)
  implicit none
  integer,intent(in):: n,m
  real(8),intent(in):: a(n)
  real(8),intent(in):: b(m)
  real(8),intent(out):: c(n,m)

  integer:: i,j

  write(*,*)'n,m=',n,m

  do i=1,m
     do j=1,n

        c(j,i) = a(j)*b(i)

     enddo
  enddo

end subroutine vavbmc



subroutine apb(n,a,b,c)
  implicit none
  integer,intent(in):: n
  real(8),intent(in):: a(n,n)
  real(8),intent(in):: b(n,n)
  real(8),intent(out):: c(n,n)

  integer:: i,j


  do i=1,n
     do j=1,n

        c(j,i) = a(j,i)+b(j,i)

     enddo
  enddo

end subroutine apb


subroutine apb_p(p,n,a,b,c)
  implicit none
  integer,intent(in):: n
  real(8),intent(in):: p
  real(8),intent(in):: a(n,n)
  integer,intent(in):: b(n,n)
  real(8),intent(out):: c(n,n)

  integer:: i,j

  write(*,*)'int(b)'
  write(*,'(4I10)')b

  do i=1,n
     do j=1,n

        c(j,i) = (a(j,i)+b(j,i))*p 

     enddo
  enddo

  c= c*p 

end subroutine apb_p


end module m_mylib 


